Remotion/Core/Core/Context/SafeContext.Timers.cs

```diff
@@ -0,0 +1,57 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+using System;
+using System.Threading;
+using System.Timers;
+
+namespace Remotion.Context
+{
+  public static partial class SafeContext
+  {
+    /// <summary>
+    /// Provides methods for setting a callback on <see cref="System.Timers.Timer"/> which are <see cref="SafeContext"/> aware.
+    /// </summary>
+    public static class Timers
+    {
+      public static ElapsedEventHandler AddElapsedEventHandler (System.Timers.Timer timer, ElapsedEventHandler handler)
```

Comment:
Since you can't write
```
      System.Timers.Timer t = new();
      var x = t.Elapsed += (sender, args) => { };
```
I suggest we make the result void here and in line 39?

Reply:
okay, I see now why this is designed this way. There is no good way to deregister the event handler without getting a reference to it. I'm not 100% convided that the scenario is helpful/needed for practical applications. If someone really does need this, they can still do it themselves, but for the most part, there's no registration/deregistration need and us returning the new handler-instance may confuse the intended usage scenarios. thoughts?

Reply:
We can do it either way but it is not easy to notice that the handler is being changed when you get a void result. And if remove the handler you might not notice that it did nothing and is still being called. Having a return type for the funtion makes that explicit and forces the consumers to make a conscious decision.

Reply:
I would add documentation that explains why a value is returned. The return type should force them to think about why something is returned, and they can look to the docs for the answer. This allows them to make their own decision if they want to keep the delegate or not.